File: backend/src/ToDoList.Core/Dtos/CreateTodoListRequest.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace ToDoList.Core.Dtos
{
    public class CreateTodoListRequest
    {
        [Required]
        [MaxLength(200)]
        public string Name { get; set; } = string.Empty;

        public int SortOrder { get; set; }

        /// <summary>When the user made this change (UTC). Conflict arbiter; server uses UtcNow if null.</summary>
        public DateTime? ClientModifiedDate { get; set; }
    }
}

File: clients/android/app/src/main/java/com/yeplist/app/data/remote/dto/CreateTodoListRequest.kt
package com.yeplist.app.data.remote.dto

data class CreateTodoListRequest(
    val name: String,
    val sortOrder: Int = 0,
    // Real user-edit time (ISO-8601 UTC). Newest-edit-wins conflict arbiter.
    val clientModifiedDate: String? = null
)
